client/scripts/controllers/chain/substrate/collective_proposal.ts

```diff
@@ -107,14 +107,27 @@ export class SubstrateCollectiveProposal
     this._Accounts = Accounts;
     this._Collective = Collective;
     this._call = eventData.call;
-    this._title = formatCall(eventData.call);
+    this.title = entity.title || formatCall(eventData.call);
     this.createdAt = entity.createdAt;
 
     entity.chainEvents.forEach((e) => this.update(e));
 
-    this._initialized = true;
-    this.updateVoters();
-    this._Collective.store.add(this);
+    if (!this._completed) {
+      const slug = chainEntityTypeToProposalSlug(entity.type);
```

Comment:
Assuming this is the same for every `Proposal` build on chain events (and noting that the `completed` flag is also general to the proposal), you may want to abstract this into a call in the top-level `Proposal` class rather than copy it into every specific subclass.